cmd/soroban-cli/src/commands/contract/bindings/typescript.rs

```diff
@@ -61,22 +67,41 @@ pub enum Error {
     Wasm(#[from] wasm::Error),
     #[error("Failed to get file name from path: {0:?}")]
     FailedToGetFileName(PathBuf),
+    #[error("cannot parse contract ID {0}: {1}")]
+    CannotParseContractId(String, DecodeError),
+    #[error("Missing RPC Url")]
+    MissingRpcUrl,
+    #[error(transparent)]
+    UtilsError(#[from] get_spec::Error),
 }
 
-impl Cmd {
-    pub async fn run(&self) -> Result<(), Error> {
+#[async_trait::async_trait]
+impl NetworkRunnable for Cmd {
+    type Error = Error;
+    type Result = ();
+
+    async fn run_against_rpc_server(
+        &self,
+        global_args: Option<&global::Args>,
+        _config: Option<&config::Args>,
+    ) -> Result<(), Error> {
         let spec = if let Some(wasm) = &self.wasm {
             let wasm: wasm::Args = wasm.into();
             wasm.parse()?.spec
         } else {
-            let fetch = contract::fetch::Cmd {
```

Comment:
It would be convenient if we had a `spec` command, that outputted just the spec as XDR / JSON, given a network and contract id, network and wasm hash, or wasm file, and then if we used that here too like we were using the fetch command. Maybe some clean up that we could follow up with.

Reply:
Good point! Could even add the schema!